### PR TITLE
Change tmpdir fixture to tmpdir_factory (#3095)

### PR DESCRIPTION
**Pull request by lilyminium**
Master version of #3095 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: testsuite/MDAnalysisTests/formats/test_libdcd.py [hunk 1]
```diff
@@ -322,14 +323,18 @@ def write_dcd(in_name, out_name, remarks='testing', header=None):
                     and sys.maxsize <= 2**32) or
                     platform.machine() == 'aarch64',
                    reason="occasional fail on 32-bit windows and ARM")
+# occasionally fails due to unreliable test timings
+@hypothesis.settings(deadline=None)  # see Issue 3096
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/formats/test_libdcd.py:327]: Does this remove the need for the xfail above?

Comment (lilyminium) [reply testsuite/MDAnalysisTests/formats/test_libdcd.py:327]: Hmm let's give it a try

Comment (IAlibay) [reply testsuite/MDAnalysisTests/formats/test_libdcd.py:327]: I guess we're still not testing on ARM64, so it'd just be the 32bit Windows stuff.

**Comment by orbeckst** [comment]
Did the 3.8 https://github.com/MDAnalysis/mdanalysis/pull/3097/checks?check_run_id=1724495378 fail because if issues in the citation processing?

**Comment by lilyminium** [comment]
@orbeckst possibly, I think it may be an intermittent error?

File: testsuite/MDAnalysisTests/formats/test_libdcd.py (hunk 1 above)
Comment (lilyminium) [reply testsuite/MDAnalysisTests/formats/test_libdcd.py:327]: @IAlibay no idea why but it makes the Travis Py2.7 test fail again, so the xfail can be left for `develop` instead....

**Pull request merged**